### DISCUSSION TITLE
Inquiry on Generating Time-Varying CDL Channels

### DISCUSSION THREAD

**Discussion by hssam1002** [NVlabs/sionna#634] [Q&A]
Hi, I am trying to generate channels using Sionna for various simulations. This is my first time using Sionna, so I am having some difficulty creating time-varying channels. I would like to clarify a couple of points on generating time-varying channel samples by setting batch_size and num_time_steps:

Does setting num_time_steps result in a time-varying channel of that length? For instance, if num_time_steps is set to 14 * 100, will the output be 1400 OFDM symbols (i.e., 100 TTI) that follow Jake's model for time correlation?

Does batch_size generate that many independent batches of channels, each following the same settings?

**Comment by SebastianCa** [comment]
Hi @hssam1002,

yes, that's correct. The `batch_size` defines independent realizations of the channel (following the same statistics).
The `num_time_steps` is the number of consecutive (OFDM) symbols from the TDL/CDL/RT model (depending which channel model you are using). Note that the length of a TTI (etc.) depends on your exact system configuration.

**Reply by hssam1002** [reply]
Thanks Sebastian. My curiosity has been resoloved :)
